src/runtime/components/ChatMessages.vue

```diff
@@ -1,14 +1,14 @@
 <script lang="ts">
 import type { AppConfig } from '@nuxt/schema'
-import type { UIMessage, ChatStatus } from 'ai'
+import type { UIMessage, ChatStatus, UIDataTypes, UITools } from 'ai'
 import theme from '#build/ui/chat-messages'
 import type { ButtonProps, ChatMessageProps, ChatMessageSlots, IconProps } from '../types'
 import type { ComponentConfig } from '../types/tv'
 
 type ChatMessages = ComponentConfig<typeof theme, AppConfig, 'chatMessages'>
 
-export interface ChatMessagesProps {
-  messages?: UIMessage[]
+export interface ChatMessagesProps<METADATA = unknown, DATA_PARTS extends UIDataTypes = UIDataTypes, TOOLS extends UITools = UITools> {
+  messages?: UIMessage<METADATA, DATA_PARTS, TOOLS>[]
```

Comment:
```suggestion
export interface ChatMessagesProps<T extends UIMessage[] = UIMessage[]> {
  messages?: T
```

Typescript should be able to infer the underlying `METADATA`, `DATA_PARTS`, `TOOLS`